cli/module_registry.go

```diff
@@ -235,6 +236,18 @@ func UploadModuleAction(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
+	tarballPath := moduleUploadPath
+	if !isTarball(tarballPath) {
+		tarballPath, err = createTarballForUpload(moduleUploadPath, c.App.Writer)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err := os.Remove(tarballPath); err != nil {
```

Comment:
Defer UncheckedErrorFunc wasn't much cleaner. Open to suggestions.

Reply:
If you aren't attached to logging an error from inability to remove, you could use [`utils.RemoveFileNoError`](https://github.com/viamrobotics/rdk/blob/main/utils/file.go#L29). We use that to remove generated `.sock` files in module management code. It also only removes the file if it exists.